### ISSUE TITLE
Not working on chrome after some time

### ISSUE THREAD

**Issue by praksahsri11** [inspirit/jsfeat#53]
I am checking on chrome latest version. when i run sample orb it is working fine after some time it is creating Aw! Snap error.. When i check on windows task manager it is utilizing the full memory. Can you help me to solve this.

**Comment by inspirit** [comment]
can reproduce this issue, it seems to be smth related to chrome itself i suppose

**Comment by praksahsri11** [comment]
ya. it is problem with the chrome only. older versions and firefox and all working fine. even in chrome when i load the page its working fine, the problem comes only when the image data starts processing it keeps pushing the data in the memory and when gets loaded we get that error.
Any work around for this ?

**Comment by inspirit** [comment]
sorry i corrected my answer i can NOT see the issue, it works just fine for me

**Comment by praksahsri11** [comment]
can you try in any windows based operating system like windows 7 or 8 only chrome browser.

**Comment by PseudoSky** [comment]
## My Guess

The links on the demo site are relative

```html
<a href="sample_orb.html" target="_blank" role="button" class="btn btn-mini">ORB MATCHING DEMO</a>
```

If you end up at the demo site without ssl, then navigate to a demo, it will still be http.
Since "getUserMedia() no longer works on insecure origins." you will get 

<img width="884" alt="screen shot 2016-01-09 at 4 57 16 pm" src="https://cloud.githubusercontent.com/assets/3356993/12218608/34f1d00a-b6f2-11e5-97a1-e6a06f3ba8fc.png">

with the error

<img width="554" alt="screen shot 2016-01-09 at 4 57 34 pm" src="https://cloud.githubusercontent.com/assets/3356993/12218614/40e7f696-b6f2-11e5-8f56-395e9f385eaa.png">

Navigating to https:

<img width="878" alt="screen shot 2016-01-09 at 4 58 20 pm" src="https://cloud.githubusercontent.com/assets/3356993/12218617/52b19bac-b6f2-11e5-9f33-639d146c5f6a.png">

This happens for me on both chrome and firefox. Firefox doesn't seem to tell you the appropriate error in the console.

## Options?

Make demo site links non relative
```html
<a href="https://inspirit.github.io/jsfeat/sample_orb.html" target="_blank" role="button" class="btn btn-mini">ORB MATCHING DEMO</a>
```

Mention the https requirement for user media?

**Comment by inspirit** [comment]
@PseudoSky i think you are using old version of repo. i've changed everything to https a while ago.
the issue reported by @praksahsri11 is smth different

**Comment by PseudoSky** [comment]
Oh hmm, I was running them from the demo site

https://inspirit.github.io

http://inspirit.github.io

**Comment by Nateowami** [comment]
@inspirit Could you update the links on http://inspirit.github.io/jsfeat to use https? I doubt most people would realize the problem was that it isn't https. I almost didn't think to switch. I know the readme links to the https version of the documentation, but people coming from NPM or other sites will find the examples broken.

**Comment by inspirit** [comment]
I don't have access to NPM repo, it was created by someone else.
here on github i've changed all the links to be https.

**Comment by Nateowami** [comment]
Yeah, I know you don't have access to the NPM package, but the links on http://inspirit.github.io/jsfeat are not https. They're protocol-dependent. Anyone who reaches the page on http will find the demos broken, with a "WebRTC not available" error message.

**Comment by inspirit** [comment]
well rather than changing all links to absolute (which is not right) I would ask the NPM author to change only one link to make it work actually. i understand that it makes it look like it is broken but this kind of change doesn't feel right to me.

**Comment by Nateowami** [comment]
@inspirit If you go to https://github.com/inspirit/jsfeat/settings and check "Enforce HTTPS" it will redirect HTTP requests to HTTPS.

**Comment by inspirit** [comment]
@Nateowami cool! that was really helpful, thank you! i've changed the setting.

**Comment by Nateowami** [comment]
@inspirit I didn't even know it was possible until right now. They announced it in June: https://github.com/blog/2186-https-for-github-pages

**Comment by leegee** [comment]
Sadly, that didn't solve the problem, and none of the demos run when viewing the GitHub pages, at least from Chrome on Windows. There are two other tickets commenting on this (#85 and #84).
